relayer-cli/src/commands/create/connection.rs

```diff
@@ -0,0 +1,159 @@
+use abscissa_core::{Command, Options, Runnable};
+
+use ibc::ics02_client::state::ClientState;
+use ibc::ics24_host::identifier::{ChainId, ClientId};
+use ibc::Height;
+use ibc_relayer::config::StoreConfig;
+use ibc_relayer::connection::Connection;
+use ibc_relayer::foreign_client::ForeignClient;
+
+use crate::commands::cli_utils::{spawn_chain_runtime, ChainHandlePair, SpawnOptions};
+use crate::conclude::{exit_with_unrecoverable_error, Output};
+use crate::prelude::*;
+
+#[derive(Clone, Command, Debug, Options)]
+pub struct CreateConnectionCommand {
+    #[options(
+        free,
+        required,
+        help = "identifier of the side `a` chain for the new connection"
+    )]
+    chain_a_id: ChainId,
+
+    #[options(free, help = "identifier of the side `b` chain for the new connection")]
+    chain_b_id: Option<ChainId>,
+
+    #[options(help = "delay period parameter for the new connection", no_short)]
+    delay: Option<u64>,
+
+    #[options(
+        help = "identifier of client hosted on chain `a`; default: None (creates a new client)",
+        no_short
+    )]
+    client_a: Option<ClientId>,
+
+    #[options(
+        help = "identifier of client hosted on chain `b`; default: None (creates a new client)",
+        no_short
+    )]
+    client_b: Option<ClientId>,
+}
+
+// cargo run --bin hermes -- create connection ibc-0 ibc-1
+// cargo run --bin hermes -- create connection ibc-0 ibc-1 --delay 1000
+// cargo run --bin hermes -- create connection ibc-0 --client-a-id 07-tendermint-0 --client-b-id 07-tendermint-0 [--delay <delay>]
+impl Runnable for CreateConnectionCommand {
+    fn run(&self) {
+        match &self.chain_b_id {
+            Some(side_b) => self.create_from_scratch(side_b),
+            None => self.create_reusing_clients(),
+        }
+    }
+}
+
+impl CreateConnectionCommand {
+    /// Creates a connection as well as brand-new clients on each side.
+    fn create_from_scratch(&self, chain_b_id: &ChainId) {
+        let config = app_config();
+
+        let spawn_options = SpawnOptions::override_store_config(StoreConfig::memory());
+
+        let chains =
+            ChainHandlePair::spawn_with(spawn_options, &config, &self.chain_a_id, chain_b_id)
+                .unwrap_or_else(exit_with_unrecoverable_error);
```

Comment:
This approach was an experiment, but I like the outcome.

Introducing the `exit_with_unrecoverable_error` and trait `Unrecoverable` allows to keep code more concise in the CLI, for example:

```diff
-        let chains = match ChainHandlePair::spawn_with(
-            spawn_options,
-            &config,
-            chain_a_id,
-            chain_b_id,
-        ) {
-            Ok(chains) => chains,
-            Err(e) => return Output::error(format!("{}", e)).exit(),
-        };
+      let chains = ChainHandlePair::spawn_with(spawn_options, &config, chain_a_id, chain_b_id)
+                 .unwrap_or_else(exit_with_unrecoverable_error);
```

@romac @ancazamfir: I'm curious what you think about it. The name `Unrecoverable` sounds a lot like a Java interface, but nothing better came up at first glance.

Reply:
I don't mind the name at all. I am all in favor for all tricks that reduce the code size and make it more readable.